metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/enrichment/MetisRecordParser.java

```diff
@@ -43,31 +43,45 @@
 
   @Override
   public Set<SearchTermContext> parseSearchTerms(RDF rdf) {
-    final List<ProxyType> providerProxies = RdfProxyUtils.getProviderProxies(rdf);
-    final Map<FieldValue, Set<FieldType>> result = new HashMap<>();
-    for (FieldType field : FieldType.values()) {
-      providerProxies.stream().map(field::extractFieldValuesForEnrichment)
-              .flatMap(Collection::stream)
-              .forEach(value -> result.computeIfAbsent(value, key -> new HashSet<>()).add(field));
+    //Proxy search terms
+    final List<AboutType> providerProxies = RdfEntityUtils.getProviderProxies(rdf).stream()
+        .map(AboutType.class::cast).collect(Collectors.toList());
+    final Set<SearchTermContext> resultSearchTermsSet = getFieldValueSet(ProxyFieldType.values(),
+        providerProxies);
+    final List<AboutType> aggregations = rdf.getAggregationList().stream()
+        .map(AboutType.class::cast).collect(Collectors.toList());
+    resultSearchTermsSet.addAll(getFieldValueSet(AggregationFieldType.values(), aggregations));
+    return resultSearchTermsSet;
+  }
+
+  private Set<SearchTermContext> getFieldValueSet(FieldType[] fieldTypes,
+      List<AboutType> aboutTypes) {
+    final Map<FieldValue, Set<FieldType>> fieldValueFieldTypesMap = new HashMap<>();
+    for (FieldType fieldType : fieldTypes) {
+      aboutTypes.stream().map(fieldType::extractFieldValuesForEnrichment)
+          .flatMap(Collection::stream).forEach(
+          value -> fieldValueFieldTypesMap.computeIfAbsent(value, key -> new HashSet<>())
+              .add(fieldType));
     }
-    return result.entrySet().stream().map(entry -> new SearchTermContext(entry.getKey().getValue(),
-            entry.getKey().getLanguage(), entry.getValue())).collect(Collectors.toSet());
+    return fieldValueFieldTypesMap.entrySet().stream().map(
+        entry -> new SearchTermContext(entry.getKey().getValue(), entry.getKey().getLanguage(),
+            entry.getValue())).collect(Collectors.toSet());
```

Comment:
I think this whole chunk could potentially become more elegant when changing the generic typing as suggested for FieldType.

Reply:
Updated.